pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -152,14 +183,25 @@ func (r *ReconcilePerconaServerMongoDB) Reconcile(request reconcile.Request) (re
 		RequeueAfter: r.reconcileIn,
 	}
 
+	// As operator can handle a few clusters
+	// lock should be created per cluster to not lock cron jobs of other clusters
+	l, ok := r.lockers.Load(request.NamespacedName.String())
+	if !ok {
+		l = lock{
+			statusMutex: new(sync.Mutex),
+			updateSync:  new(int32),
+		}
+		r.lockers.Store(request.NamespacedName.String(), l)
+	}
```

Comment:
Should we put both ops behind the same lock, e.g. inside Load? If two or more instance Reconcile for the same cluster would be running for the first time "simultaneously" they could rewrite locks of each other not instead of loading. I don't think that is possible now since there Reconcile is not being run concurrently, but anyway...

Or! If Reconcile always runs sequentially, let's abandon a synchronisation on "lockers" at all. No need to synchronize access to "lockers" between Reconcile and Cron. Look:
No lock for the cluster at that point means it is the first run hence cronjob for that cluster yet. So, create just a lock and add it to the map. And the corn closure shouldn't load the lock on each run but capture the loaded lock from the environment (see. my comments for `r.crons.crons.AddFunc()`). Therefore no concurrently run Reconcile + no lock loads by the cron closure in go-routine - no need for the mutex.

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/472/commits/9a8a46cec1d2a2523f4f9cf9fe19fb31147d5ca4